services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/northbound/switches/SwitchRulesSpec.groovy

```diff
@@ -0,0 +1,102 @@
+package org.openkilda.functionaltests.spec.northbound.switches
+
+import static org.junit.Assume.assumeTrue
+import static org.openkilda.testing.Constants.WAIT_OFFSET
+
+import org.openkilda.functionaltests.BaseSpecification
+import org.openkilda.functionaltests.helpers.Wrappers
+import org.openkilda.messaging.command.switches.DeleteRulesAction
+import org.openkilda.messaging.payload.flow.FlowState
+import org.openkilda.testing.model.topology.TopologyDefinition
+import org.openkilda.testing.service.lockkeeper.LockKeeperService
+import org.openkilda.testing.service.northbound.NorthboundService
+
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.beans.factory.annotation.Value
+
+class SwitchRulesSpec extends BaseSpecification {
+
+    @Autowired
+    TopologyDefinition topology
+    @Autowired
+    NorthboundService northboundService
+    @Autowired
+    LockKeeperService lockKeeperService
+
+    @Value('${reroute.delay}')
+    int rerouteDelay
+    @Value('${discovery.timeout}')
+    int discoveryTimeout
+
+    def "Default rules are installed on a new switch when connecting it to the controller"() {
+        requireProfiles("virtual")
+
+        given: "A switch with no rules installed and not connected to the controller"
+        def sw = topology.getActiveSwitches().first()
+        def defaultRulesSorted = northboundService.getSwitchRules(sw.dpId).flowEntries.sort { it.cookie }
+        northboundService.deleteSwitchRules(sw.dpId, DeleteRulesAction.DROP_ALL)
+        assert Wrappers.wait(WAIT_OFFSET) { northboundService.getSwitchRules(sw.dpId).flowEntries.isEmpty() }
+
+        lockKeeperService.knockoutSwitch(sw.dpId)
+        assert Wrappers.wait(WAIT_OFFSET) { !(sw.dpId in northboundService.getActiveSwitches()*.switchId) }
+
+        when: "Connect the switch to the controller"
+        lockKeeperService.reviveSwitch(sw.dpId)
+
+        then: "The switch is really connected to the controller"
+        Wrappers.wait(WAIT_OFFSET) { sw.dpId in northboundService.getActiveSwitches()*.switchId }
+
+        and: "Default rules are installed on the switch"
+        def actualRulesSorted = northboundService.getSwitchRules(sw.dpId).flowEntries.sort { it.cookie }
+        actualRulesSorted.size() == defaultRulesSorted.size()
+        [0..actualRulesSorted.size() - 1].each { i ->
```

Comment:
There is a more pretty way to loop through 2 lists
```
[actualRules, defaultRules].transpose().each { actual, expected ->
  actual.cookie == expected.cookie
  //etc.
}
```

Reply:
Thanks a lot!